src/platform/user/authentication/utilities.js

```diff
@@ -275,14 +281,19 @@ export function sessionTypeUrl({
         codeChallengeMethod,
         ...(gaClientId && { gaClientId }),
         ...(scope && { scope }),
-        ...(queryParams.operation && { operation: queryParams.operation }),
+        ...(queryParams.operation && {
+          operation: queryParams.operation,
+        }),
+        ...(queryParams.redirect && { redirect: queryParams.redirect }), // Ensure redirect is passed
```

Comment:
We can revert the `redirect` here as our redirects are handled via the `authReturnUrl` in Session Storage